lib/polarshift/request.rb

```diff
@@ -307,6 +305,16 @@ def update_caseruns(project_id, run_id, updates)
         )
       end
 
+      def update_caseruns_smart(project_id, run_id, updates, timeout: 360)
+        res = update_caseruns(project_id, run_id, updates)
+        if res[:exitstatus] == 202
+          wait_op(url: JSON.load(res[:response])["operation_result_url"],
+                timeout: timeout)
```

Comment:
I think we shouldn't wait here. 

We rather need to repeat the operation a few times in case we get a server error 5xx. Otherwise we will not solve the issue reported in JIRA.

Reply:
@akostadinov   Updated.  Repeat the operation via wait_for, rather than wait the operation to finish.